spark-connector/spark-connector/src/main/scala/org/apache/spark/sql/iceberg/catalyst/parser/GravitinoIcebergSparkSqlExtensionsParser.scala

```diff
@@ -0,0 +1,398 @@
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.spark.sql.iceberg.catalyst.parser
+
+import com.datastrato.gravitino.spark.connector.iceberg.SparkIcebergTable
+import org.antlr.v4.runtime._
+import org.antlr.v4.runtime.atn.PredictionMode
+import org.antlr.v4.runtime.misc.{Interval, ParseCancellationException}
+import org.antlr.v4.runtime.tree.TerminalNodeImpl
+import org.apache.iceberg.common.DynConstructors
+import org.apache.iceberg.spark.ExtendedParser.RawOrderField
+import org.apache.iceberg.spark.{ExtendedParser, Spark3Util}
+import org.apache.spark.sql.{AnalysisException, SparkSession}
+import org.apache.spark.sql.catalyst.{FunctionIdentifier, TableIdentifier}
+import org.apache.spark.sql.catalyst.analysis.{EliminateSubqueryAliases, UnresolvedRelation}
+import org.apache.spark.sql.catalyst.expressions.Expression
+import org.apache.spark.sql.catalyst.parser.ParserInterface
+import org.apache.spark.sql.catalyst.parser.extensions.{IcebergParserUtils, IcebergSqlExtensionsAstBuilder, IcebergSqlExtensionsBaseListener, IcebergSqlExtensionsLexer, IcebergSqlExtensionsParser}
+import org.apache.spark.sql.catalyst.parser.extensions.IcebergSqlExtensionsParser.{NonReservedContext, QuotedIdentifierContext}
+import org.apache.spark.sql.catalyst.plans.logical._
+import org.apache.spark.sql.catalyst.trees.Origin
+import org.apache.spark.sql.connector.catalog.{Table, TableCatalog}
+import org.apache.spark.sql.execution.command.ExplainCommand
+import org.apache.spark.sql.internal.{SQLConf, VariableSubstitution}
+import org.apache.spark.sql.types.{DataType, StructType}
+
+import scala.collection.JavaConverters._
+import java.util.Locale
+import scala.util.Try
+
+/**
+ * when parsing RowLevelCommands, IcebergSparkSqlExtensionsParser only supports the Iceberg SparkTable
+ * instead of the SparkIcebergTable defined by the Gravitino spark-connector,
+ * which will cause unexpected exceptions if run RowLevelCommands using the Gravitino spark-connector.
+ * Therefore it is necessary to rewrite IcebergSparkSqlExtensionsParser,
+ * so that RowLevelCommands can run normally using the Gravitino spark-connector.
+ *
+ * GravitinoIcebergSparkSqlExtensionsParser will be injected automatically in GravitinoIcebergSparkSessionExtensions,
+ * and GravitinoIcebergSparkSessionExtensions will be automatically registered in GravitinoDriverPlugin.
+ * Injecting GravitinoIcebergSparkSessionExtensions manually is not recommended,
+ * because we need to ensure GravitinoIcebergSparkSessionExtensions is injected before IcebergSparkSessionExtensions
+ * to avoid some unexpected exceptions.
+ */
+// Referred from v3.4/spark-extensions/src/main/scala/org/apache/spark/sql/catalyst/parser/extensions/IcebergSparkSqlExtensionsParser.scala
+class GravitinoIcebergSparkSqlExtensionsParser(delegate: ParserInterface) extends ParserInterface with ExtendedParser {
```

Comment:
I wondered could we reuse the abilities of `IcebergSparkSqlExtensionsParser` and only override `isSparkIcebergTable` not copying most logic from `IcebergSparkSqlExtensionsParser`. WDYT?

Reply:
cc @FANNG1

Reply:
If we have no doubts about using scala, I will continue to optimize this. Looking forward to your reply, Thank you.

Reply:
please wait a while

Reply:
please go ahead, thanks

Reply:
updated.